### DISCUSSION TITLE
How can I run these kind of code block?

### DISCUSSION THREAD

**Discussion by shravanngoswamii** [quarto-dev/quarto-cli#9719] [Q&A] [answered]
### Description

I am converting to turing.jl's website to quarto from `.jmd` files and in one notebook it have this code block:
\```julia; fig_ext=".png"
corner(
    chain,
    MCMCChains.namesingroup(chain, :coefficients_versicolor);
    label=[string(i) for i in 1:4],
)
\```

\```julia; fig_ext=".png"
corner(
    chain,
    MCMCChains.namesingroup(chain, :coefficients_virginica);
    label=[string(i) for i in 1:4],
)
\```
old `.jmd` file: https://github.com/TuringLang/TuringTutorials/blob/9f8fface50f5e067d6f78b89c1da87ac2adcc9f2/tutorials/08-multinomial-logistic-regression/08_multinomial-logistic-regression.jmd#L139C1-L153C4

I want to implement it here in `.qmd` file: https://github.com/TuringLang/TuringTutorials/blob/c998e960ccf4b975e532dbd42b8598f613a7942a/tutorials/08-multinomial-logistic-regression/index.qmd#L141C1-L157C4

**Comment by mcanouil** [comment]
I suggest to take a look at the get started guides on authoring with Quarto (https://quarto.org/docs/get-started/) and to look at the Julia guide (https://quarto.org/docs/computations/julia.html)

**Reply by shravanngoswamii** [reply]
I read them but does not seem to find anything for this: ```julia; fig_ext=".png"

**Reply by cscheid** [reply]
Because that's not syntax we support. Unfortunately we can't write documentation about other software packages in our own.

**Reply by shravanngoswamii** [reply]
I did not meant that, I am trying to execute those code blocks and they are giving error because there is no option for this parameter `fig_ext="png"` in Quarto, I tried `#| fig-format: png` but its also not working!

**Reply by mcanouil** [reply]
`fig-format` is a project or front matter level option, not a code cell option.

**Reply by shravanngoswamii** [reply]
ok I will look for some another solution but can you also tell me if `#| warning: true or false` option works with `engine: julia` or not?

**Comment by shravanngoswamii** [comment]
Does Quarto support building qmd files in parallel?
our current buildkite workflow builds multiple jmd files in parallel
So, if one jmd file fails execution, the other jmd notebooks still get compiled correctly.
Can you tell whether there is a way to build qmd notebooks in parallel within a project?

**Reply by cscheid** [reply]
> Can you tell whether there is a way to build qmd notebooks in parallel within a project?

There isn't one currently.

**Comment by shravanngoswamii** [comment]
Is there any way to render this table in normal format, this table is being generated from this codeblock:
\```{julia}
\# Import the "iris" dataset.
data = RDatasets.dataset("datasets", "iris");

\# Show twenty random rows.
data[rand(1:size(data, 1), 20), :]
\```

| I am getting this | I want this |
|--------|--------|
| ![Screenshot 2024-05-21 002954](https://github.com/quarto-dev/quarto-cli/assets/123811742/d10b3aac-8dee-4297-90d6-dce05e8677ca) | ![Screenshot 2024-05-21 003348](https://github.com/quarto-dev/quarto-cli/assets/123811742/e43f1f8a-7d6e-4dcd-87c4-603e2de8a500) |

**Reply by shravanngoswamii** [reply]
I want my table output to look like they look in terminals

**Reply by mcanouil** [reply]
Note that we don't know what a "normal format" is.

Use a library that allows you to customise your table in the appropriate language for your formats.
For instance, "write" HTML/CSS for HTML based format.
Note that Quarto can read/parse HTML tables into other formats as long as Pandoc knows how to write those other formats.

Customisation is up to you using standard technologies.
Quarto is very unlikely going to change default style.

**Reply by shravanngoswamii** [reply]
from "normal format", I meant the table that is coming from execution of any code block should should look like they look in terminal

**Reply by mcanouil** [reply]
> should look like they look in terminal

I am afraid not.
Quarto chose to have consistent table output computed in code cells.

Note that Quarto uses what exists in terms of styling.
A table in Julia has no formatting data, it's entirely controlled by Julia when processed. Your object has no styling whatsoever.
If you want styling, then provide one.

**Reply by shravanngoswamii** [reply]
Well, How can I give styling to it's output? You can see in the left image that how bad its formatting is!

**Reply by mcanouil** [reply, marked as answer]
> You can see in the left image that how bad its formatting is!

That's your point of view.

You need to find the libraries and use them, e.g., PrettyTables.jl.
Note that's not related to Quarto.
Formatting of computed outputs is really a work that you have to do yourself using the libraries of your chosen computing language, especially if you don't like the default.
By the way, that's not specific to Quarto or even Jupyter.

**Reply by shravanngoswamii** [reply]
ok
